Fix Pagination

mzagaja opened the pull request:
Resolve MAPC/datacommon#117 by changing the pageCount and length to reference the data in the selected rows computed property instead of the entire raw dataset. This makes it so that the user now goes to the correct page when clicking the end of pages.

mzagaja committed:
Fix Pagination

Resolve MAPC/datacommon#117 by changing the pageCount and length to reference the data in the selected rows computed property instead of the entire raw dataset. This makes it so that the user now goes to the correct page when clicking the end of pages.

ericyoungberg reviewed (dismissed):
The paginator breaks if you remove years from the year filter while you are past the new max page number. The paginator should check the new bounds and fix your current page if necessary.

<img width="1225" alt="capture d ecran 2018-08-17 a 10 13 49 am" src="https://user-images.githubusercontent.com/6749126/44270761-6868c400-a206-11e8-9d82-e88a9bc96e3d.png">

mzagaja committed:
Fix Empty Page Problem

This resolves the issue of being shown and left on an empty page in the browser if you go to the end of the page list and then remove a year from the list by recalculating the max and minimum data rows to display when the year action is toggled.

iangmhill reviewed (approved):
LGTM
